schema/image_test.go

```diff
@@ -9,6 +9,13 @@ func Test_BuildImageName_DefaultFormat(t *testing.T) {
 	if got != want {
 		t.Errorf("BuildImageName want: \"%s\", got: \"%s\"", want, got)
 	}
+
+	want = "foo:123d/img:latest"
```

Comment:
I would like to see new tests for new scenarios.

by the way, `foo:123d/img:latest` looks like an invalid image tag. 🤔

Reply:
in this case that produce the correct, because for that test it is using the `default` and not the `--tag sha` option

Reply:
`foo:123d` looks like an invalid prefix to me

Reply:
i was trying to reproduce the item described in the issue


```
$ faas-cli build --tag=sha --lang python3 --image foo:4567/bar --handler . --name my-function
[...]
Successfully tagged foo:4567/bar-52f0c55:latest
Image: foo:4567/bar-52f0c55 built.
```

the `foo:123d` can be whatever I guess